dracut/30ignition/ignition-complete.target

```diff
@@ -7,9 +7,11 @@
 Description=Ignition Complete
 Before=initrd.target
 
-# Make sure if any ExecStart= fail, the boot fails. This normally would
-# already be guaranteed by `initrd.target` failing, but that doesn't seem to be
-# enough: https://bugzilla.redhat.com/show_bug.cgi?id=1696796
+# initrd.target has OnFailureJobMode=replace-irreversibly, which seems to
+# cause unit restart loops in the initramfs if one of our units fails.  Thus
+# we intercept Ignition unit failures here and isolate to emergency.target.
+# Some context in https://bugzilla.redhat.com/show_bug.cgi?id=1696796
+# but the test case in that bug no longer works on F30 and F31.
```

Comment:
Ok. The text here contains all of the content I'm looking for but I think we can make it more obvious to the user. i.e. we need to TELL someone when they can remove this workaround.

Let's answer these two questions:

- How *should* things behave?
- When can I remove this workaround?

In this case here is my guess:

- My attempt at an answer to how *should* things behave?
    - initrd.target has `OnFailureJobMode=replace-irreversibly` which means that any failures in our units should lead us to emergency.target and the boot should isolate there in a failure state waiting on input from the user.
- My attempt at an answer to when can I remove this?
    - In the future when openbug1234 is fixed we should be able to remove the two `OnFailure*` statements below and leave it up to initrd.target to catch the error and send us into emergency.target.

I have no idea if my above attempts at an answer are correct. They are just my interpretations of your text.

Reply:
Filed https://github.com/coreos/ignition-dracut/issues/115 with more details and updated the comment with a link to it.